[vcpkg] fix X_VCPKG_APPLOCAL_DEPS_INSTALL

dhrdlicka opened the pull request:
**Describe the pull request**
This pull request fixes applocal.ps1 getting invoked with double `CMAKE_INSTALL_PREFIX` in install targets when `X_VCPKG_APPLOCAL_DEPS_INSTALL` is enabled:
```
C:/Users/David/Documents/hello/build/cmake_install.cmake(49):  message(-- Installing app dependencies for hello... )
-- Installing app dependencies for hello...
C:/Users/David/Documents/hello/build/cmake_install.cmake(50):  execute_process(COMMAND C:/Program Files/WindowsApps/Microsoft.PowerShell_7.1.1.0_x64__8wekyb3d8bbwe/pwsh.exe -noprofile -executionpolicy Bypass -file C:/Users/David/AppData/Local/Programs/vcpkg/scripts/buildsystems/msbuild/applocal.ps1 -targetBinary ${CMAKE_INSTALL_PREFIX}/C:/Program Files (x86)/Hello/bin/hello.exe -installedDir C:/Users/David/Documents/hello/build/vcpkg_installed/x64-windows/debug/bin -OutVariable out )
```

dhrdlicka committed:
[vcpkg] fix X_VCPKG_APPLOCAL_DEPS_INSTALL

JackBoosY commented on scripts/buildsystems/vcpkg.cmake:
```diff
@@ -553,7 +553,7 @@ if(X_VCPKG_APPLOCAL_DEPS_INSTALL)
             set(PARSED_TARGETS "")
 
             # Destination - [RUNTIME] DESTINATION argument overrides this
-            set(DESTINATION "${CMAKE_INSTALL_PREFIX}/bin")
+            set(DESTINATION "bin")
```
I think we should use other methods to deal with this situation. We always need `CMAKE_INSTALL_PREFIX` here.

dhrdlicka replied:
Using it there breaks CPack and other scenarios that might override the prefix at install time such as `cmake --install build --prefix <custom prefix>`.

ras0219-msft replied:
Using the local path is correct here.

dhrdlicka committed:
[vcpkg] fix x_vcpkg_install_local_dependencies

Fixed `x_vcpkg_install_local_dependencies` always assuming a relative `DESTINATION`.

dhrdlicka commented:
It seems like I overlooked `x_vcpkg_install_local_dependencies` always assuming a relative destination path, which was the real cause behind the double prefix bug I tried to fix. Added a check for that in the new commit.

JackBoosY commented:
cc @sandercox for review this PR.

strega-nil commented:
/azp run

ras0219-msft commented:
LGTM, thanks for the fix!
